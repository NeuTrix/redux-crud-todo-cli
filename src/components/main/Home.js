import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
// @material-ui
import { withStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';

import cssLogo from '../../assets/mediaCardLogos/resized_images/css_logo.png';
import htmlLogo from '../../assets/mediaCardLogos/resized_images/html_logo.png';
import expressLogo from '../../assets/mediaCardLogos/resized_images/exp.png';
import jsLogo from '../../assets/mediaCardLogos/resized_images/js_logo.png';
import MediaCard from '../buttons/MediaCard';
import mongoLogo from '../../assets/mediaCardLogos/resized_images/mongo_logo.png';
import nodeLogo from '../../assets/mediaCardLogos/resized_images/node_logo.png';
import reactLogo from '../../assets/mediaCardLogos/resized_images/react_logo.png';
import reduxLogo from '../../assets/mediaCardLogos/resized_images/redux_logo.png';

const propTypes = {
	authorized: PropTypes.bool.isRequired,
	classes: PropTypes.instanceOf(Object).isRequired,
};

const Home = ({ authorized, classes }) => {
	const loginButton = (
		<NavLink to="/login">
			<Button
				className={classes.button}
				color="primary"
				component="button"
				variant="contained"
			>
				{'Login'}
			</Button>
		</NavLink>
	);

	const registerButton = (
		<NavLink to="/register">
			<Button
				className={classes.button}
				color="primary"
				component="button"
				type="submit"
				variant="contained"
			>
				{'Register'}
			</Button>
		</NavLink>
	);

	return (
		<div className={classes.grid}>

			<Typography
				className={classes.mainTitle}
				variant="h4"
			>
				{'Welcome to Done'}
			</Typography>

			<Typography
				className={classes.subTitle}
				variant="h6"
			>
				{'a full-stack React CRUD app'}
			</Typography>

			<div className={classes.tilesDisplay}>
				<div className={classes.tiles}>
					<MediaCard imageUrl={mongoLogo} label="MongoDB" />
				</div>
				<div className={classes.tiles}>
					<MediaCard imageUrl={nodeLogo} label="Node" />
				</div>
				<div className={classes.tiles}>
					<MediaCard
						imageUrl={expressLogo}
						width="150%"
						label="express"
					/>
				</div>
				<div className={classes.tiles}>
					<MediaCard imageUrl={reactLogo} label="React" />
				</div>
				<div className={classes.tiles}>
					<MediaCard imageUrl={reduxLogo} label="Redux" />
				</div>
				<div className={classes.tiles}>
					<MediaCard imageUrl={htmlLogo} label="HTML5" />
				</div>
				<div className={classes.tiles}>
					<MediaCard imageUrl={cssLogo} width="75px" label="CSS3" />
				</div>
				<div className={classes.tiles}>
					<MediaCard imageUrl={jsLogo} width="100px" label="JavaScript" />
				</div>
			</div>

			<div className={classes.buttonDisplay}>
				<div>{ !authorized && loginButton }</div>
				<div>{ !authorized && registerButton }</div>
			</div>

		</div>
	);
};

const styles = () => ({
	// login and register buttons
	button: {
		color: 'white',
		width: 100,
		margin: 20,
	},

	buttonDisplay: {
		// background: 'lime',
		display: 'flex',
		gridArea: 'buttonDisplay',
		justifyContent: 'center',
		paddingTop: 25,
		width: 'inherit',
	},
	// The media card display section
	tilesDisplay: {
		// background: 'orangered',
		display: 'flex',
		flexWrap: 'wrap',
		placeContent: 'center',
		gridArea: 'tilesDisplay',
		marginTop: 25,
	},

	grid: {
		display: 'grid',
		gridTemplateAreas: `
			"mainTitle"
			"subTitle"
			"tilesDisplay"
			"buttonDisplay"
		`,
		placeItems: 'center',
		maxWidth: 800,
	},

	login: {
		gridArea: 'login',
	},

	mainTitle: {
		gridArea: 'mainTitle',
	},

	subTitle: {
		gridArea: 'subTitle',
	},

	tiles: {
		// border: '3px solid purple',
		float: 'left',
		margin: 10,
		minWidth: 150,
	},

});

Home.propTypes = propTypes;

export default withStyles(styles)(Home);
