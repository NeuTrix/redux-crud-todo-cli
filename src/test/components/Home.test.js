
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from 'enzyme';
import Home from '../../components/main/Home';

describe('The Home component', () => {
  const wrapper = render(
    <Router>
      <Home/>
    </Router>
  );

  it('...renders correctly', () => {
    expect(wrapper).toMatchSnapshot({
      class: expect.any(String)
    });
  });

});