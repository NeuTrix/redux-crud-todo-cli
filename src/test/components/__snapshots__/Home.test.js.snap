// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[object Object] 1`] = `
<div
  class="Home engrBox paper sc-bdVaJa gfIpsM"
>
  <div
    class="_title ctr engr under sc-bxivhb jLpweO"
  >
    <h1>
      React-Todo (beta)
    </h1>
  </div>
  <div
    class="-subtitle ctr sc-ifAKCX fTxVjk"
  >
    <span>
      Fullstack MERN CRUD web app featuring:
    </span>
  </div>
  <button
    class="undefined box ctr mat sc-EHOje dvyVgq"
  >
    Mongo
  </button>
  <button
    class="undefined box ctr mat sc-bZQynM bDrDIS"
  >
    Express
  </button>
  <button
    class="undefined box ctr mat sc-gzVnrw gSXYkK"
  >
    React
  </button>
  <button
    class="undefined box ctr mat sc-htoDjs bOeUTW"
  >
    NodeJS
  </button>
  <button
    class="undefined box ctr mat sc-dnqmqq hyOftF"
  >
    Redux
  </button>
  <button
    class="undefined box ctr mat sc-iwsKbI fhhztn"
  >
    Grid
  </button>
</div>
`;
